lib/elixir/lib/enum.ex

```diff
@@ -2358,13 +2358,17 @@ defmodule Enum do
 
   ## Examples
 
-      iex> Enum.with_index [:a, :b, :c]
+      iex> Enum.with_index([:a, :b, :c])
       [a: 0, b: 1, c: 2]
 
+      iex> Enum.with_index([:a, :b, :c], 3)
+      [a: 3, b: 4, c: 5]
+
   """
-  @spec with_index(t) :: [{element, non_neg_integer}]
-  def with_index(enumerable) do
-    map_reduce(enumerable, 0, fn x, acc ->
+  @spec with_index(t) :: [{element, integer}]
+  def with_index(enumerable), do: with_index(enumerable, 0)
+  def with_index(enumerable, offset) do
```

Comment:
@GeorgeTaveras1231 technically it is completely the same what you've done, but obviously less code and that's the common practice throughout codebase (I believe it is true for every Elixir project), thus there is no reason not to use it. :bowtie: